app/views/granules/download.sh.erb

```diff
@@ -1,20 +1,88 @@
 #!/bin/sh
 
+<% if @urls.first.start_with? 'https://datapool.asf.alaska.edu' %>
+set_cookie ()
+{
+    TOKEN=`curl -s -d "userid=$1&password=$2" https://vertex.daac.asf.alaska.edu/services/authentication -i | grep Set-Cookie | cut -d ' ' -f 2 | cut -d '=' -f 2 | sed 's/;$//'`
+    echo ".alaska.edu	TRUE	/	FALSE	0	datapool	$TOKEN" > ~/.asf_cookies
+    chmod 0600 ~/.asf_cookies
+}
+
+if [ $# -eq 2 ]; then
+    set_cookie $1 $2
+    echo "\n Cookie file has been successfully set in your home directory. Re-run the script without any arguments."
+    exit
+fi
+
 if command -v curl >/dev/null 2>&1
 then
+    if [ ! -f ~/.asf_cookies ]; then
+        echo "\n.asf_cookies file not found. Please configure your username and password for authentication using a .asf_cookies file:\n    > $0 <uid> <password>\nwhere <uid> is your Earthdata Login username and <password> is your Earthdata Login password."
+        exit
+    fi
+
     fetch_urls() {
-        while read -r line; do curl -Og $line; done;
+        while read -r line; do curl -b ~/.asf_cookies -Og -L $line; done;
     }
+elif command -v wget >/dev/null 2>&1
+then
+    if [ ! -f ~/.asf_cookies ]; then
+        echo "\n.asf_cookies file not found. Please configure your username and password for authentication using a .asf_cookies file:\n    > cd ~\n    > wget --keep-session-cookies --save-cookies ~/.asf_cookies --post-data='userid=<uid>&password=<password>' --no-check-certificate https://vertex.daac.asf.alaska.edu/services/authentication\n    > chmod 0600 .asf_cookies\nwhere <uid> is your Earthdata Login username and <password> is your Earthdata Login password."
+        exit
+    fi
 
+    fetch_urls() {
+        while read -r line; do wget --load-cookies ~/.asf_cookies --save-cookies ~/.asf_cookies --keep-session-cookies $line; done;
+    }
+else
+    fail_with_error "Error: Could not find a command-line downloader.  Please install curl or wget"
+fi
+
+<% else %>
+
+if command -v curl >/dev/null 2>&1
+then
+    STATUS_CODE=`curl -s -z "$(date)" -w %{http_code} <%= raw @urls.first %> | tail -1`
+    if [ "$STATUS_CODE" -ne "200" || "$STATUS_CODE" -ne "304" ]; then
+        # Need URS authentication
+        if [ ! -f ~/.netrc ]; then
+            echo "\n.netrc file not found. Please configure your username and password for authentication using a .netrc file:\n    > echo 'default login <uid> password <password>' > .netrc\nwhere <uid> is your Earthdata Login username and <password> is your Earthdata Login password."
+            exit
+        fi
+
+        APPROVED_STATUS=`curl -s -b ~/.urs_cookies -c ~/.urs_cookies -L --max-redirs 2 -n <%= raw @urls.first %> -w %{http_code} | tail  -1`
+        if [ "$APPROVED_STATUS" -ne "302" ]; then
```

Comment:
Different provider returns different status code here. However, if it is approved, a ```302``` will be responded.